### ISSUE TITLE
object_usage_linter should warn about missing packages

### ISSUE THREAD

**Issue by AshesITR** [r-lib/lintr#2872]
When getting Namespace exports for library calls fails, this is currently silent. This can cause confusing lints for scripts, e.g.

```r
library(tibble) # and lintr is run where tibble is not installed

f <- function() {
  tibble(x = 42L)
}
```

In practice, this can easily occur when working with `renv` and pre-commit because the latter pulls packages from the system library by default instead of from a local `renv`

**Comment by MichaelChirico** [comment]
One thing about `warning` is it may not be surfaced to the user (e.g. if the backend process of the IDE ignores any condition < error). Maybe it could go in the lint message instead? `NB: execution found calls to attach %d libraries that were not found by lintr: tibble, ...`

**Comment by AshesITR** [comment]
The lint message would be repeated many times in a real script, though.

Maybe produce a separate lint on the problematic library call?
